app/src/test/java/org/mozilla/fenix/settings/quicksettings/DefaultQuickSettingsControllerTest.kt

```diff
@@ -0,0 +1,207 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.quicksettings
+
+import android.app.Activity
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import androidx.navigation.NavController
+import assertk.assertAll
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isTrue
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import io.mockk.verifyOrder
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.ObsoleteCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import mozilla.components.browser.session.Session
+import mozilla.components.feature.session.SessionUseCases
+import mozilla.components.feature.sitepermissions.SitePermissions
+import mozilla.components.feature.tabs.TabsUseCases
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.IntentReceiverActivity
+import org.mozilla.fenix.TestApplication
+import org.mozilla.fenix.browser.BrowserFragment
+import org.mozilla.fenix.components.PermissionStorage
+import org.mozilla.fenix.exceptions.ExceptionDomains
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@UseExperimental(ObsoleteCoroutinesApi::class)
+@RunWith(RobolectricTestRunner::class)
+@Config(application = TestApplication::class)
+class DefaultQuickSettingsControllerTest {
+    private val context = testContext
+    private val coroutinesScope = GlobalScope
+    private val navController = mockk<NavController>(relaxed = true)
+    private val browserSession = mockk<Session>()
+    private val permissionStorage = mockk<PermissionStorage>(relaxed = true)
+    private val trackingExceptions = mockk<ExceptionDomains>(relaxed = true)
+    private val reload = mockk<SessionUseCases.ReloadUrlUseCase>(relaxed = true)
+    private val addNewTab = mockk<TabsUseCases.AddNewTabUseCase>(relaxed = true)
+    private val requestPermissions = mockk<(Array<String>) -> Unit>()
+    private val displayPermissions = mockk<() -> Unit>(relaxed = true)
+    private val dismiss = mockk<() -> Unit>(relaxed = true)
```

Comment:
This is interesting.  I would have expected this to bleed state between tests, but I tested it and it doesn't.

So 1) it's super cool that Mockk takes care of that for us (I've been using `MockkAnnotations#init` in an `@Before` block, and I guess I don't need to), and 2) I really wish their docs went into more depth.

Reply:
Are you referring to only these methods from lines 53-56?
Don't think there is any state of them to bleed. Just one off method calls.
Been a little lucky to have no state to be careful to clean up in this tests. :)

Reply:
Specifically, I meant the `every` blocks in each method, since we're setting these mocks up once and sharing them for all of the tests.  But I tested it and couldn't produce any errors.  My guess would be that MockK resets these between tests.